### ISSUE TITLE
Attributes resulting of an unpacked tuple are not documented (Python)

### ISSUE THREAD

**Issue by splint33** [doxygen/doxygen#9947]
Hello everyone, I am currently using Doxygen 1.9.7 (b4e95d34ecf85c5e77a897650f56da86199cd7ba*).

I am trying to sanitize my project documentation, and I am reviewing all warnings returned by doxygen.
While doing so, I have encountered an issue.

On a Python project, I have a class with private attributes.
Thoses attributes are initialized with a value resulting of an unpacked tuple.
When generating the documentation, only the private attribute "_parameter" is documented, attributes "_A" to "_E" are not present.

```
##	@class		XXX
##	@brief		XXX
##	
##	@details	XXX
class myClass:

	# ---------------------- Constructor ----------------------

	##	@brief		XXX
	##	@param		XXX
	##	
	##	@details	XXX
	def __init__(self, a_parameter):
		## Internal variable holding a reference to XXX
		self._parameter= a_parameter
		## Those attributes will not be processed !
		(self._A, self._B, self._C, self._D, self._E) = myFunctionReturningATuple(self._parameter)
```

I have tried to remove the parenthesis on the attributes.
By doing so, only the last attribute is referenced into the documentation, and not the four previous ones.

**Comment by albert-github** [comment]
The problem here is indeed the fact that a tuple is returned, which is not supported at the moment.
Question here is also what should be shown in this case?

Note: for the future please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.
This will make it easier to reproduce the problem.

**Comment by splint33** [comment]
Oh, I didn't know tuples where not supported.

I thought that it will work like when the doxyfile parameter "DISTRIBUTE_GROUP_DOC" is set to YES, generating the same documentation for all attributes.

**Comment by doxygen** [comment]
@splint33 Please verify if the referenced commit fixes the problem for you. Do not close this issue, as this will be done when the next official version is released.

**Comment by albert-github** [comment]
@doxygen looks like `self._E` is not documented:

**Comment by albert-github** [comment]
Looks like it is even a bit worse with:
```
# Another class 6
class Test6:
    ## Document tuple vars
    (self._A, self._D, self._E) = myFunctionReturningATuple(self._testRunner)

    ## constructor
    def __init__(self):
      ## Document tuple vars in init
      (self._AA, self._AD, self._AE) = myFunctionReturningATuple(self._testRunner)
```

**Comment by doxygen** [comment]
@albert-github I've fixed the problem with the last argument in the list. The `Test6` example looks like invalid Python, i.e. it uses `self` outside of a method.

**Comment by albert-github** [comment]
Yes the `self._A` are probably invalid python.

```
## Another class 7
class Test7:
    ## constructor
    def __init__(self):
      self._testRunner = 42
      ## Document tuple vars in init
      (self._AA, self._AD, self._AE) = myFunctionReturningATuple(self._testRunner)
```
this documents `_testRunner` also as "Document tuple vars in init", I don't think this is intended.
When `_testRunner` has documentation the "(in case `_testRunner` is documented the documented value will be the " will be merged with it.

**Comment by doxygen** [comment]
@albert-github I agree that the part at the right of the assignment should not be considered. Fixed that now.

**Comment by albert-github** [comment]
Now a tuple outside a class:
```
## \file

## a fie
def myFunctionReturningATuple():
    return(41,42,43)

## Document tuple vars outside
(AA, AD, AE) = myFunctionReturningATuple()

print("%d %d %d"%(AA, AD, AE))
```
only `AA` is documented.

**Comment by doxygen** [comment]
@albert-github Also addressed the global tuple case

**Comment by albert-github** [comment]
Tuple outside the class looks OK.

New example:
```
## \file

## first global
global cnt
## second global
global cnt1

cnt = 42
cnt1 = 142

## a fie
def fie():
    global cnt
    cnt = cnt+1
    return cnt

## a fie
def fie1():
    global cnt1
    cnt1 = cnt1+1
    return (cnt1)

## var 1
A = fie()
## var 2
(B) = fie()
## var 3
C = fie1()
## var 4
(D) = fie1()

print("%d %d %d %d"%(A, B, C, D))
```
results in:
![image](https://user-images.githubusercontent.com/5223533/227763559-5a29be2e-d025-4de1-818b-1dd306d01db3.png)


1. global `cnt1` not documented and intended documentation added to `cnt`
2. `fie()` / `fie1()` in front of `A` / `C`, looks like a separate issue in the 1.9.6 version the functions and also the `A` / `C` had the word `def` in front now this is gone for the functions (#9846 / #9848) but apparently the variables still got the the `def` and now the function name as the type (that the tuple variable is / was OK might be a coincidence).

**Comment by doxygen** [comment]
@albert-github Indeed, these are not tuple related. 

Not sure documenting a "global var" makes much sense as it is typically used to import an existing variable as a global into a local scope (i.e. to prevent a local alias). So the fact that doxygen doesn't see this as a thing that can be documented is not really a problem in my opinion.

**Comment by albert-github** [comment]
I think we can conclude that the problems I saw regarding tuples are solved and thus the problems from this issue.

Regarding the "global var" problem, a bad example but valid python:
```
## \file

## the init fie
def init_fie():
   ## lets document it as global
   global cnt_fie
   ## lets document it in assign
   cnt_fie = 42

## the compute fie
def compute_fie():
   ## lets document it in compute
   global cnt_fie
   return cnt_fie


init_fie()
print("%d"%(compute_fie()))
```

but there is no documentation found for `cnt_fie` ...

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.7.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
